services/black/src/ambassador/BlackClient.cs

```diff
@@ -1,70 +1,80 @@
+    public class BlackClient : IDisposable
     {
         private EndpointConfiguration senderConfiguration;
+        private bool disposedValue;
+        private IEndpointInstance? senderEndpoint;
 
-        public BlackClient(string connectionString)
+        public BlackClient(string rabbitMqConnectionString)
         {
             senderConfiguration = new EndpointConfiguration("Black.Ambassador");
-            //senderConfiguration.SendOnly();
-            // ToDo I don't like this callback stuff with nservicebus, we could just use a webapi
-            senderConfiguration.EnableCallbacks();
-            senderConfiguration.EnableInstallers();
-            senderConfiguration.MakeInstanceUniquelyAddressable($"Black.Ambassador.{Guid.NewGuid()}");
-            senderConfiguration.UseSerialization<NewtonsoftSerializer>();
+            senderConfiguration
+                .ConfigureDefaults(
+                    rabbitMqConnectionString, 
+                "Black.Api", 
+                    typeof(SetGuestInformation), typeof(GuestInformationRequest))
+                .WithCallbacks($"Black.Ambassador.{Guid.NewGuid()}");
+        }
 
-            var conventions = senderConfiguration.Conventions();
+        public async Task StartAsync()
+        {
+            this.senderEndpoint = await Endpoint.Start(senderConfiguration).ConfigureAwait(false);
+        }
 
-            conventions.DefiningCommandsAs(t =>
-                t.Namespace != null &&
-                t.Namespace.EndsWith(".Commands") &&
-                !t.Name.EndsWith("Response"));
+        public async Task SetGuestInformationAsync(Guid orderId, GuestInformation guestInformation)
+        {
+            ThrowIfNotInitialized();
+            var sgi = new SetGuestInformation(orderId, guestInformation);
+            if (!SetGuestInformationValidator.IsValid(sgi))
+                throw new InvalidOperationException($"{typeof(GuestInformation).Name} is invalid.");
 
-            conventions.DefiningMessagesAs(t =>
-                t.Namespace != null &&
-                t.Namespace.EndsWith(".Commands") &&
-                t.Name.EndsWith("Response"));
+            await senderEndpoint.Send(sgi).ConfigureAwait(false);
+        }
 
-            conventions.DefiningEventsAs(t =>
-                t.Namespace != null &&
-                t.Namespace.EndsWith(".Events"));
+        public async Task<GuestInformation?> GetGuestInformationAsync(Guid orderId)
+        {
+            ThrowIfNotInitialized();
+            var response = await senderEndpoint.Request<GuestInformationResponse>(new GuestInformationRequest { OrderId = orderId }).ConfigureAwait(false);
+            return response.GuestInformation;
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    // ToDo maybe use NetStandard2.1 (no .NetFramework support!) and IAsyncDisposable
```

Comment:
`IAsyncDisposable` for sure!
We can go with netstandard2.1 or even netcoreapp3.1 since these projects are scoped only to our repo.